src/MongoDB.Driver.Core/Core/Misc/ServerVersion.cs

```diff
@@ -0,0 +1,433 @@
+        /// <returns>True if the string representation was parsed successfully; otherwise false.</returns>
+        public static bool TryParse(string value, out ServerVersion result)
+        {
+            if (!string.IsNullOrEmpty(value))
+            {
+                var pattern = @"(?<major>\d+)\.(?<minor>\d+)(\.(?<patch>\d+)(-(?<preRelease>.*))?)?";
+                var match = Regex.Match((string)value, pattern);
+                if (match.Success)
+                {
+                    var major = int.Parse(match.Groups["major"].Value);
+                    var minor = int.Parse(match.Groups["minor"].Value);
+                    var patch = match.Groups["patch"].Success ? int.Parse(match.Groups["patch"].Value) : 0;
+                    var preRelease = match.Groups["preRelease"].Success ? match.Groups["preRelease"].Value : null;
+
+                    result = new ServerVersion(major, minor, patch, preRelease);
+                    return true;
+                }
+            }
+
+            result = null;
+            return false;
+        }
+
+        // public operators
+        /// <summary>
+        /// Determines whether two specified server versions have the same value.
+        /// </summary>
+        /// <param name="a">The first server version to compare, or null.</param>
+        /// <param name="b">The second server version to compare, or null.</param>
+        /// <returns>
+        /// True if the value of a is the same as the value of b; otherwise false.
+        /// </returns>
+        public static bool operator ==(ServerVersion a, ServerVersion b)
+        {
+            if (object.ReferenceEquals(a, null))
+            {
+                return object.ReferenceEquals(b, null);
+            }
+
+            return a.CompareTo(b) == 0;
+        }
+
+        /// <summary>
+        /// Determines whether two specified server versions have different values.
+        /// </summary>
+        /// <param name="a">The first server version to compare, or null.</param>
+        /// <param name="b">The second server version to compare, or null.</param>
+        /// <returns>
+        /// True if the value of a is different from the value of b; otherwise false.
+        /// </returns>
+        public static bool operator !=(ServerVersion a, ServerVersion b)
+        {
+            return !(a == b);
+        }
+
+        /// <summary>
+        /// Determines whether the first specified ServerVersion is greater than the second specified ServerVersion.
+        /// </summary>
+        /// <param name="a">The first server version to compare, or null.</param>
+        /// <param name="b">The second server version to compare, or null.</param>
+        /// <returns>
+        /// True if the value of a is greater than b; otherwise false.
+        /// </returns>
+        public static bool operator >(ServerVersion a, ServerVersion b)
+        {
+            if (object.ReferenceEquals(a, null))
```

Comment:
what the difference between this method and methods below? Why they are different?

Reply:
What do you mean different?

Some of the methods are implemented by calling others for code reuse, if that's what you mean.

Reply:
Correction: I did not insist on a DIFFERENT implementation. I only requested that the EXISTING implementation not be changed because there was not enough reason to do so.

Reply:
>if that's what you mean.

yep, I didn't notice it originally